File: backend/app/api/generate.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.schemas.generation import (
    GenerateRequest,
    SpriteSheetRequest,
    GenerationResponse,
)
from app.services.generation_service import create_generation

router = APIRouter()


@router.post("/generate", response_model=GenerationResponse)
async def generate_image(req: GenerateRequest, db: AsyncSession = Depends(get_db)):
    gen = await create_generation(
        db=db,
        prompt=req.prompt,
        model=req.model,
        provider=req.provider,
        negative_prompt=req.negative_prompt,
        aspect_ratio=req.aspect_ratio,
        image_size=req.image_size,
        transparent_bg=req.transparent_bg,
        reference_image_b64=req.reference_image_b64,
    )
    return gen


@router.post("/generate/sprite-sheet", response_model=GenerationResponse)
async def generate_sprite_sheet(
    req: SpriteSheetRequest, db: AsyncSession = Depends(get_db)
):
    gen = await create_generation(
        db=db,
        prompt=req.prompt,
        model=req.model,
        provider=req.provider,
        negative_prompt=req.negative_prompt,
        aspect_ratio=req.aspect_ratio,
        image_size=req.image_size,
        transparent_bg=req.transparent_bg,
        is_sprite_sheet=True,
        sprite_config=req.sprite_config.model_dump(),
        reference_image_b64=req.reference_image_b64,
    )
    return gen

File: frontend/src/api/generate.ts
import client from "./client";
import type {
  GenerateRequest,
  SpriteSheetRequest,
  Generation,
  UploadResponse,
} from "../types";

export async function generateImage(req: GenerateRequest): Promise<Generation> {
  const { data } = await client.post<Generation>("/generate", req);
  return data;
}

export async function generateSpriteSheet(
  req: SpriteSheetRequest
): Promise<Generation> {
  const { data } = await client.post<Generation>(
    "/generate/sprite-sheet",
    req
  );
  return data;
}

export async function uploadReference(file: File): Promise<UploadResponse> {
  const formData = new FormData();
  formData.append("file", file);
  const { data } = await client.post<UploadResponse>(
    "/upload-reference",
    formData,
    { headers: { "Content-Type": "multipart/form-data" } }
  );
  return data;
}
